src/PowerShellEditorServices/Services/PowerShell/Debugging/DscBreakpointCapability.cs

```diff
@@ -1,45 +1,37 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.Extensions.Logging;
-using Microsoft.PowerShell.EditorServices.Logging;
-using Microsoft.PowerShell.EditorServices.Services.DebugAdapter;
 using System;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
+using System.Linq;
 using System.Management.Automation;
 using System.Threading;
-using SMA = System.Management.Automation;
-using Microsoft.PowerShell.EditorServices.Services.PowerShell.Utility;
-using Microsoft.PowerShell.EditorServices.Services.PowerShell.Runspace;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Services.DebugAdapter;
+using Microsoft.PowerShell.EditorServices.Services.PowerShell.Execution;
 using Microsoft.PowerShell.EditorServices.Services.PowerShell.Host;
+using Microsoft.PowerShell.EditorServices.Services.PowerShell.Runspace;
 
 namespace Microsoft.PowerShell.EditorServices.Services.PowerShell.Debugging
 {
     internal class DscBreakpointCapability
     {
+        private static bool? isDscInstalled;
         private string[] dscResourceRootPaths = Array.Empty<string>();
+        private readonly Dictionary<string, int[]> breakpointsPerFile = new();
 
-        private readonly Dictionary<string, int[]> breakpointsPerFile =
-            new();
-
-        public async Task<BreakpointDetails[]> SetLineBreakpointsAsync(
+        public async Task<IEnumerable<BreakpointDetails>> SetLineBreakpointsAsync(
             IInternalPowerShellExecutionService executionService,
             string scriptPath,
-            BreakpointDetails[] breakpoints)
+            IEnumerable<BreakpointDetails> breakpoints)
         {
-            List<BreakpointDetails> resultBreakpointDetails =
-                new();
-
             // We always get the latest array of breakpoint line numbers
             // so store that for future use
-            if (breakpoints.Length > 0)
+            if (breakpoints.Any())
```

Comment:
So I don't think it's particularly impactful here but it's worth noting that double enumeration will happen for the first item. Granted the only code being executed twice is `BreakpointDetails.Create` but it adds more surface for easy bugs as if at some point `Create` is no longer "pure" then we'd get unintended side effects.

Reply:
How do you avoid that?